lib/cartopy/mpl/geoaxes.py

```diff
@@ -642,7 +641,7 @@ def set_extent(self, extents, crs=None):
         # extents are obviously the same as the projection domain.
         try_workaround = ((crs is None and
                            isinstance(self.projection, ccrs.PlateCarree)) or
-                          crs == self.projection)
+                          isinstance(crs, ccrs.PlateCarree))
```

Comment:
What if `crs` _isn't_ PlateCarree?

Reply:
Yeah good point.  Didn't think that through, sorry.  I'll try again.